pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -155,3 +189,55 @@ func (nc *nodeConfig) handleCSR(requestorFilter string) error {
 
 	return nil
 }
+
+// GetNode returns a pointer to the node object associated with the external IP provided
+func (nc *nodeConfig) getNode(externalIP string) (*v1.Node, error) {
+	var matchedNode *v1.Node
+
+	nodes, err := nc.k8sclientset.CoreV1().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get list of nodes")
+	}
+	if len(nodes.Items) == 0 {
+		return nil, fmt.Errorf("no nodes found")
+	}
+
+	// Find the node that has the given IP
+	for _, node := range nodes.Items {
+		for _, address := range node.Status.Addresses {
+			if address.Type == "ExternalIP" && address.Address == externalIP {
+				matchedNode = &node
+				break
+			}
+		}
+		if matchedNode != nil {
+			break
+		}
+	}
+	if matchedNode == nil {
+		return nil, fmt.Errorf("could not find node with IP: %s", externalIP)
+	}
+	return matchedNode, nil
+}
+
+// waitForHybridOverlayAnnotation waits for the hybrid overlay subnet annotation to be present on the node until the
+// timeout is reached
+func (nc *nodeConfig) waitForHybridOverlayAnnotation(nodeName string) error {
+	var found bool
+	err := wait.Poll(retry.Interval, retry.Timeout, func() (bool, error) {
+		node, err := nc.k8sclientset.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error getting node %s: %v", nodeName, err)
+		}
+		_, found := node.Annotations[HybridOverlaySubnet]
+		if found {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	if !found {
+		return errors.Wrap(err, fmt.Sprintf("timeout waiting for %s node annotation", HybridOverlaySubnet))
+	}
```

Comment:
I dont think found will ever be true at this point. If it was true it would have returned within the `poll`
You can remove the if statement and just return error here.

Reply:
I believe you're right, my mistake